### ISSUE TITLE
dnf package resource version comparison failures, part 2

### ISSUE THREAD

**Issue by epilatow** [chef/chef#10950]
In PR #9847 I made [a comment](https://github.com/chef/chef/pull/9847/files#r540720136) where I pointed out that we're still calling the broken `versioncompare` function with identical strings (so things were working by accident). Well, today someone asked me with an dnf resource `:upgrade` failure that I believe is being caused by these broken version comparison calls with non-equal strings. The resource in questions is:
```
package 'sudo' do
  action :upgrade
end
```

The currently installed rpm version is 0:1.8.29-6.el8, and the latest available version is 0:1.8.29-6.el8_3.1. Here's what we see in the chef debug logs:
```
INFO: Processing dnf_package[sudo] action upgrade (fbit_sudo::default line 35)
DEBUG: sending '{"action":"whatinstalled","provides":"sudo"}' to python helper
DEBUG: got 'sudo 0:1.8.29-6.el8 x86_64' from python helper
DEBUG: parsed sudo-0:1.8.29-6.el8.x86_64 from python helper
DEBUG: sending '{"action":"whatinstalled","provides":"sudo"}' to python helper
DEBUG: got 'sudo 0:1.8.29-6.el8 x86_64' from python helper
DEBUG: parsed sudo-0:1.8.29-6.el8.x86_64 from python helper
DEBUG: sending '{"action":"whatavailable","provides":"sudo"}' to python helper
DEBUG: got 'sudo 0:1.8.29-6.el8_3.1 x86_64' from python helper
DEBUG: parsed sudo-0:1.8.29-6.el8_3.1.x86_64 from python helper
DEBUG: sending '{"action":"versioncompare","versions":["0:1.8.29-6.el8.x86_64","0:1.8.29-6.el8_3.1.x86_64"]}' to python helper
DEBUG: got '0' from python helper
DEBUG: dnf_package[sudo] sudo 0:1.8.29-6.el8_3.1.x86_64 is already installed
```

In this case, since the EVRA parsing code in `version_tuple()` assumes that an architecture is always found after the first `.` after the first `-` in the version string the two version strings above will be parse to the same EVR tuple of (E = 0; V = 1.8.29; R = 6).

**Comment by epilatow** [comment]
Looking at a chef run from a server, I also see we're still passing in EVRs and EVRAs into versioncompare, there by making it impossible know how to correctly parse the input (since we can't differentiate between the two). Here's a grep of log results removing EVRA to EVRA comparisons
```
# grep -a versioncompare chef.20210127.1815.1611800129.out | \
    sed -e 's/^.* sending //' | \
    egrep -v "\.x86_64.*\.x86_64|\.noarch.*\.noarch"
'{"action":"versioncompare","versions":["0:8.0p1-31.fb6.el8.x86_64","8.0p1-31.fb6.el8"]}' to python helper
'{"action":"versioncompare","versions":["0:8.0p1-31.fb6.el8.x86_64","8.0p1-31.fb6.el8"]}' to python helper
'{"action":"versioncompare","versions":["0:8.0p1-31.fb6.el8.x86_64","8.0p1-31.fb6.el8"]}' to python helper
```

**Comment by epilatow** [comment]
Fixed by #10951
